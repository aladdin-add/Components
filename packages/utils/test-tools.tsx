import React from "react";
import { mount, shallow } from "enzyme";
import { ThemeProvider } from "styled-components";
import { ConfigContext } from "@//utils/context/config";
import theme from "@//theme";
const config =  require("../../orchard.theme.config.json");

const removeProperties = () => {
  const keys = ["theme", "styledTheme"];
  return {
    test: (val) => {
      return (
        val &&
        typeof val === "object" &&
        "props" in val &&
        Object.keys(val.props).some((prop) => keys.some((key) => key === prop))
      );
    },
    print: (val, serialize) => {
      keys.forEach((key) => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors.common,
    modes: theme.colors.modes,
    ...theme.colors.modes.light,
  },
};

export const shallowWithTheme = (children) => shallow(
      <ThemeProvider theme={formattedTheme}>
        {children}
      </ThemeProvider>
  )
    .dive()
    .shallow();

export const mountWithTheme = (children) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        {children}
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
