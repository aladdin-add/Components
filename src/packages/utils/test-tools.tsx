import React, { ReactNode } from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { ConfigContext } from '@/utils/context/config';
import theme from '@/theme';

const config = require('../../orchard.theme.config.json');

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: any) => {
      return (
        val &&
        typeof val === 'object' &&
        'props' in val &&
        Object.keys(val.props).some(prop => keys.some(key => key === prop))
      );
    },
    print: (val: any, serialize: any) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors.common,
    modes: theme.colors.modes,
    ...theme.colors.modes.light,
  },
};

export const shallowWithTheme = (children: ReactNode) =>
  shallow(<ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>)
    .dive()
    .shallow();

export const mountWithTheme = (children: ReactNode) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>
    </ConfigContext.Provider>
  );
};
