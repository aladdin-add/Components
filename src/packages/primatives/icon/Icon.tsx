import React, { useContext, Suspense } from 'react';
import { HelpCircle } from 'react-feather';
import { generateAutomationId } from '@/utils/automation';
import { BoxProps } from '@/primatives/box';
import { ConfigContext } from '@/utils/context/config';

import { getIcon } from './utils';
import { SVG } from './styled';

export interface Props extends Omit<BoxProps, 'children'> {
  name: string;
}

const Icon = ({ autoid, name, ...rest }: Props) => {
  const config = useContext(ConfigContext);

  const SvgElement = SVG(getIcon(config, name));

  return (
    <Suspense fallback={<HelpCircle />}>
      <SvgElement
        data-autoid={`${generateAutomationId(autoid ?? name)}_icon`}
        {...rest}
      />
    </Suspense>
  );
};

export default Icon;
