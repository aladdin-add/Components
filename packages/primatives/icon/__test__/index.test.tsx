import React from "react";
import toJson from "enzyme-to-json";
import { shallowWithTheme } from "@//test-tools";

import Icon from "..";

describe("<Icon />", () => {
  it("renders correctly", () => {
    const wrapper = shallowWithTheme(<Icon name="test" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
