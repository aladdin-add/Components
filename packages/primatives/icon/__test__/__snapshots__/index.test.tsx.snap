// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly 1`] = `
<Component>
  <Icon
    name="test"
  >
    <Suspense
      fallback={<HelpCircle />}
    >
      <SVG
        data-autoid="test_icon"
      />
      <HelpCircle>
        <svg
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
          />
          <line
            x1="12"
            x2="12.01"
            y1="17"
            y2="17"
          />
        </svg>
      </HelpCircle>
    </Suspense>
  </Icon>
</Component>
`;
